File: ie/dit/SortedLL2.java
/*	Sorted Linked List using Sentinel Node
	Kalyan Kola Cahill - C16357813
	DT228/2 - 15/02/2018
*/
package ie.dit;

class SortedLL2
{
    Node head;
    
	//Node Structure
    private static class Node{
        int data;
        Node next;
    }	
	
	//Constructor
    public SortedLL2(){
        this.head = new Node();
    }

	//Print Nodes Method
    public void display()
	{
        Node t = new Node();
		t = this.head;
        
        System.out.print("\nHead -> ");
        while(t.next != null) {
            System.out.print(t.data + " -> ");
            t = t.next;
        }
        System.out.println("End\n");
    }	
	
	//Linked List Insert Node Method
    public void insert(int x)
    {
        Node newNode = new Node();
        newNode.data = x;
		Node current = new Node();
		current = this.head;
        
        while(current.next != null && x > current.next.data)
		{
			current = current.next;
		}
		
		newNode.next = current.next;
		current.next = newNode;
		
		return;
    }

	//Linked List Remove Node Method
    public void remove(int x) 
	{
		if(this.head.next == null)
		{
			System.out.println("List is empty");
            return;
		}
		
		Node current = new Node();
		current = this.head;
		while (current.next.data != x && current.next != null)
		{
			current = current.next;	
		}
		
		if(current.next.data == x)
		{
			current.next = current.next.next;
		}
		else
		{
			System.out.println("Value not in list");
		}
		
		return;
    }
    
	//Linked List Main Method
    public static void main(String[] args)
    {
        SortedLL2 list = new SortedLL2();
        list.display();
        
        int i, x;
        
		//Inserting New Nodes and Printing List
        for(i = 1; i <= 10; i++)  
		{
            x =  (int)(Math.random()*100.0);
            System.out.println("Inserting " + x);
            list.insert(x);
            list.display();
        }
		
		//Inserting 12 and Removing
		System.out.println("Inserting 12");
		list.insert(12);
		list.display();
		System.out.println("Removing 12");
		list.remove(12);
		list.display();
    }
}

File: SortedLL2.py
##	Sorted Linked List using Sentinel Node
##	Kalyan Kola Cahill - C16357813
##	DT228/2 - 15/02/2018

class SortedLL:

    def __init__(self):
        self.head = Node(None, 'Start')
	
	#print node method
    def print_nodes(self):
        current = self.head

        while current:
            print(current.data)
            current = current.next


	#insert new node method
    def insert(self, value):
        new_node = Node(None, value)
        current = self.head

		#iterates through Linked List until new node value <= current 
		#node value
        while (current.next != None and value > current.next.data):
            current = current.next

        new_node.next = current.next
        current.next = new_node

	#remove node method
    def remove(self, value):

		#error message if list is empty
        if self.isEmpty():
            print("List is Empty")
            return

        current = self.head

		#iterates through Linked List until current node value =
		#selected node value
        while (current.next.data != value and current.next != None):
            current = current.next

        # If we found the value
        if current.next.data == value:
            current.next = current.next.next

        else:
            print("\nValue not in the list\n")



	#Linked List isEmpty method
    def isEmpty(self):
        current = self.head

        if current.next == None:
            return True
        else:
            return False

#define Node structure
class Node:
    def __init__(self, node, data):
        self.data = data
        self.next = node



sorted_list = SortedLL()

#inserting 5 unsorted nodes into sorting Linked List
sorted_list.insert(5)
sorted_list.insert(3)
sorted_list.insert(2)
sorted_list.insert(1)
sorted_list.insert(4)
sorted_list.print_nodes()
print("\n")

#removing node from sorted Linked List
sorted_list.remove(3)
sorted_list.print_nodes()
